### PR TITLE
TECH-326: Test Function Refactoring

### PR DESCRIPTION
**Pull request by parousisv**
**Summary**

This is a huge task of refactoring all test functions created under platformvm. Refactoring is mostly on error throws but some changes need to be done not only in test files.

**Purpose of change**

Most of the error handling is implemented quite generically and as a result, some test functions may catch an error that has been falsely trigger and yet the whole function's result come up as correct due to the generic error handling.

**Solution**

- Replace condition error handling with assertions to all test functions.
- Add specific errors for better handling.
- Replace generic error throws with wrapped ones and fix test functions where necessary.

### CODE REVIEW

**Comment by DerTiedemann** [comment]
@parousisv Good job so far, but there are still some ci failiures, could you have a look at them @ me once they are resolved?

File: vms/platformvm/add_subnet_validator_tx_test.go [hunk 1]
```diff
@@ -52,15 +52,14 @@ func TestAddSubnetValidatorTxSyntacticVerify(t *testing.T) {
...
+	assert.NoError(t, err)
+
 	tx.UnsignedTx.(*UnsignedAddSubnetValidatorTx).NetworkID++
 	// This tx was syntactically verified when it was created...pretend it wasn't so we don't use cache
 	tx.UnsignedTx.(*UnsignedAddSubnetValidatorTx).syntacticallyVerified = false
-	if err := tx.UnsignedTx.(*UnsignedAddSubnetValidatorTx).SyntacticVerify(vm.ctx); err == nil {
-		t.Fatal("should have errored because the wrong network ID was used")
-	}
+	err = tx.UnsignedTx.(*UnsignedAddSubnetValidatorTx).SyntacticVerify(vm.ctx)
+	// unwrapped := errors.Unwrap(err)
```
Comment (peak3d) [inline vms/platformvm/add_subnet_validator_tx_test.go:61]: Remove this line??

File: vms/platformvm/advance_time_tx_test.go [hunk 2]
```diff
@@ -426,43 +403,35 @@ func TestAdvanceTimeTxRemoveSubnetValidator(t *testing.T) {
...
-		t.Fatal(err)
-	}
-	if err := vm.internalState.(*internalStateImpl).loadPendingValidators(); err != nil {
-		t.Fatal(err)
-	}
+
+	err = vm.internalState.Commit()
+	assert.NoError(t, err)
+
+	err = vm.internalState.(*internalStateImpl).loadCurrentValidators()
```
Comment (peak3d) [inline vms/platformvm/advance_time_tx_test.go:414]: loadCurrent? or loadPending?

**Review by peak3d — COMMENTED** [review]
2 small issues.....

File: vms/platformvm/advance_time_tx_test.go (hunk 2 above)
Comment (parousisv) [reply vms/platformvm/advance_time_tx_test.go]: You are right. At this point the pending validators should have been loaded.

**Comment by c4t-ag** [comment]
@parousisv close PR? It does not match to what we agreed to....

**Pull request closed without merging**